### ISSUE TITLE
Add ability to create a account with an existing steam account

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1977]
### Is your feature request related to a problem? Please describe.

The user sign up could be cumbersome in VR like typing in your e-mail and password

### Describe the solution you'd like

A way of creating a Resonite account from a steam account for easier account creation

### Describe alternatives you've considered

Creating a account as normal

### Requesters

@epicEaston197

**Comment by shadowpanther** [comment]
Some key points:
* creating an account using just the user's Steam account
* allowing to "upgrade" said Steam-linked account by adding email and password as login method
* allowing to "upgrade" existing login-password account by linking to a Steam account and allowing to use it for login

Under the hood that would mean that a Steam account (and maybe other platforms, like a Meta/Oculus account for example) needs to be added as a valid authentication method, that a Resonite account should be allowed to be created with any one of the authentication methods and that multiple authentication methods could be linked to one Resonite account.

**Comment by Frooxius** [comment]
Yes. The way we plan to add this is just adding Steam as authentication method. SkyFrost supports multiple authentication methods internally.

This will allow to:
- Create new account, there the Steam account is the only initial method (user will be able to add password later)
- Add Steam as authentication to existing account - they can use it in place of password login

We'll likely utilize the new settings to provide UI for this.

**Comment by epicEaston197** [comment]
> Yes. The way we plan to add this is just adding Steam as authentication method. SkyFrost supports multiple authentication methods internally.
> 
> This will allow to:
> 
>     * Create new account, there the Steam account is the only initial method (user will be able to add password later)
> 
>     * Add Steam as authentication to existing account - they can use it in place of password login
> 
> 
> We'll likely utilize the new settings to provide UI for this.

That is great and fulfills the needs of my closed issue earlier thank you

**Comment by Frooxius** [comment]
Closing this one in favor of official issue: https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/4057
